src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+import React, { useEffect, useRef, useState } from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+
+import FocusTrap, { FocusTrapProps } from "./focus-trap.component";
+import { RadioButton, RadioButtonGroup } from "../../components/radio-button";
+import { ModalContext } from "../../components/modal/modal.component";
+import { Select, Option } from "../../components/select";
+
+jest.useFakeTimers();
+
+interface MockComponentProps extends FocusTrapProps {
+  isAnimationComplete?: boolean;
+  triggerRefocusFlag?: boolean;
+  tabIndex?: number;
+  children: React.ReactNode;
+  shouldFocusFirstElement?: boolean;
+}
+
+const WRAPPER_ID = "test wrapper";
+const FIRST_ELEMENT = "first element";
+const BUTTON_ONE = "Test button One";
+const BUTTON_TWO = "Test button Two";
+const RADIO_LABEL_ONE = "Radio one";
+const RADIO_LABEL_TWO = "Radio two";
+const RADIO_GROUP_ONE = "Radio group one";
+const RADIO_GROUP_TWO = "Radio group two";
+
+const tabPress = () =>
+  fireEvent.keyDown(screen.getByTestId(WRAPPER_ID), {
+    key: "Tab",
+  });
+
+const shiftTabPress = () =>
+  fireEvent.keyDown(screen.getByTestId(WRAPPER_ID), {
+    key: "Tab",
+    shiftKey: true,
+  });
+
+const focusElement = (target: HTMLElement) => {
+  target.focus();
+  expect(target).toBeFocused();
```

Comment:
We may be able to replace several of our custom matchers/test helpers with identical ones from the jest-dom library, such as the `toHaveFocus` here. That way we could reduce the complexity of our spec helpers code
```suggestion
  expect(target).toHaveFocus();
```

Reply:
Good point, I'll update it as we no longer need to handle for scenarios where the passed argument isn't a DOM element when using RTL 👍